griptape/structures/structure.py

```diff
@@ -139,6 +140,7 @@ def resolve_relationships(self) -> None:
 
     @observable
     def before_run(self, args: Any) -> None:
+        RunnableMixin.before_run(self, args)
```

Comment:
why doesnt `super().before_run(args)` work here? seems kinda wonky to have to call it like this.

Reply:
There was some wacky multiple inheritance stuff going on. Although this syntax is a bit clunkier, I think it's much more explicit as to which parent we're calling into.

Reply:
with this syntax, there is no inheritance happening though. we are bypassing it completely

Reply:
it seems like `@observable` is the root cause with `super` not working

Reply:
Agreed that observable seems to be the root cause -- somehow it is mangling the method signature. I think it is a complex issue that will probably require introducing something like https://wrapt.readthedocs.io/en/master/ to resolve.

This solution is still doing everything [by the books,](https://www.geeksforgeeks.org/python-call-parent-class-method/) just a bit lower level than super. If we can resolve the observability issues in a separate PR I'm all for reverting to super(), but I don't think it should block this PR.

Reply:
I would really like to dig into this some more while we have the chance.

Reply:
Fix implemented [here](https://github.com/griptape-ai/griptape/pull/1273). Still unclear what wrapt does that fixes this over `functools.wraps`.

Reply:
I was in favor of wrapt from the beginning out of fear, but I'm also in favor of yagni. Now we get both.

Let's not forget to add a task to learn how wrapt does its magic! (and add unit tests for ensuring presence of magic)